docs/book/src/developer/architecture/controllers/control-plane.md

```diff
@@ -264,6 +264,22 @@ The `status` object **may** define several fields:
 * `externalManagedControlPlane` - is a bool that should be set to true if the Node objects do not
   exist in the cluster. For example, managed control plane providers for AKS, EKS, GKE, etc, should
   set this to `true`. Leaving the field undefined is equivalent to setting the value to `false`.
+* `.Conditions[Paused]` - is a condition that reports if the cluster or control plane resource is paused. It should check if 'spec.paused' is set on the cluster, and for the paused annotation on the resource. This is currently optional, but will become required in future.
+  ```go
+  // Return early and set the paused condition to True if the object or Cluster
+  // is paused.
+  // We assume that the change to the object has to be written, e.g. via the
+  // patchHelper in defer.
+  if annotations.IsPaused(cluster, m) {
+	  log.Info("Reconciliation is paused for this object")
+	  conditions.MarkTrueWithNegativePolarity(m, clusterv1.PausedCondition,
+    clusterv1.ResourcePausedReason, clusterv1.ConditionSeverityInfo,
+    clusterv1.ResourcePausedMessage)
+	  return ctrl.Result{}, nil
+  }
+
+  conditions.MarkFalse(m, clusterv1.PausedCondition, clusterv1.ResourceNotPausedReason, clusterv1.ConditionSeverityInfo, "Resource is operating as expected")
```

Comment:
I'm not sure if I want a `FalseConditionWithNegativePolarity` here? Does a resource not being paused, therefore reconciling now count as a positive polarity event? 

Although, then the helper does not accept a reason or severity. 

Maybe This would be more straightforward with just `conditions.Set()`

cc @sbueringer

Reply:
I would say let's use `conditions.Set()`

Let's only provide severity + reason + message if the object is paused. Goal would be to precisely write in the message why the object is paused (e.g. because the Cluster is paused, or the object itself is paused, but we can finalize wording of that message on the implementation PR)

Reply:
Ok - I've updated the PR to do this! :) 

I'm not sure if it makes sense to have reason + message, where reason can be the cluster is paused, or that the annotation is present. What extra would we gain from adding a message?

Reply:
There are multiple reasons why a resource could be paused. One example: A Machine can be paused because the Machine itself is paused or because the Cluster is paused.

I would like to surface all the reasons why something is paused so folks know what to change to unpause (and also after which "unpause" the reconciliation continues)

Reply:
Ah yep, I understand and agree with your motivation there! I just imagined those having different reasons: 

```go
// ClusterPausedReason (Severity=Info) documents a CAPI object that is paused due to the cluster being paused (.Spec.Paused).
ClusterPausedReason = "ClusterPaused"

// AnnotationPausedReason (Severity=Info) documents a CAPI object that is paused due to the paused annotation being present.
AnnotationPausedReason = "PausedAnnotationSet"

// MachineDeploymentPausedReason (Severity=Info) documents a CAPI object that is paused due to the machine deployment being paused.
MachineDeploymentPausedReason = "MachineDeploymentPaused"
```

In this instance I'm not sure what extra useful information we might add in a message? E.g in the scenario where a cluster is paused, adding the name of the cluster?  Name of the annotation?

Reply:
Ah sorry that was confusing :D. I would use one Reason constant and then just describe in the message

Reply:
Ok, I think the latest changes should have short reason + more descriptive message!